target/linux/ath79/dts/ar7242_engenius_eap350-v1.dts

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar7242.dtsi"
+
+/ {
+	compatible = "engenius,eap350-v1", "qca,ar7242";
+	model = "EnGenius EAP350 v1";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &eth0;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "eap350-v1:blue:power";
+			gpios = <&gpio 17 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+	};
+
+	ath9k-leds {
+		compatible = "gpio-leds";
+
+		wlan {
+			label = "eap350-v1:blue:wlan";
+			gpios = <&ath9k 1 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	virtual_flash {
+		compatible = "mtd-concat";
+
+		devices = <&firmware1 &firmware2>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				compatible = "openwrt,okli";
+				label = "firmware";
+				reg = <0x0 0x0>;
+			};
+		};
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <20000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x40000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x40000 0x10000>;
+			};
+
+			partition@50000 {
+				label = "custom";
+				reg = <0x50000 0x50000>;
+				read-only;
+			};
+
+			partition@a0000 {
+				label = "loader";
+				reg = <0xa0000 0x10000>;
+				read-only;
+			};
+
+			firmware2: partition@b0000 {
+				label = "firmware2";
+				reg = <0xb0000 0xf0000>;
+			};
+
+			partition@1a0000 {
+				label = "fakeroot";
+				reg = <0x1a0000 0x10000>;
+			};
+
+			firmware1: partition@1b0000 {
+				label = "firmware1";
+				reg = <0x1b0000 0x4c0000>;
+			};
+
+			partition@670000 {
+				label = "failsafe";
+				reg = <0x670000 0x180000>;
+				read-only;
+			};
+
+			art: partition@7f0000 {
+				label = "art";
+				reg = <0x7f0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy4: ethernet-phy@4 {
+		reg = <4>;
+		eee-broken-100tx;
+		eee-broken-1000t;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	mtd-mac-address = <&art 0x0>;
+
+	phy-handle = <&phy4>;
+	phy-mode = "rgmii-id";
+
+	pll-data = <0x16000000 0x00000101 0x00001313>;
+};
+
+&mdio1 {
+	status = "disabled";
+};
+
+&pcie {
+	status = "okay";
+
+	ath9k: wifi@168c,002a {
```

Comment:
This should have wifi@0,0 syntax.

Reply:
@xdarklight I did this according to the [docs you wrote](https://www.kernel.org/doc/Documentation/devicetree/bindings/net/wireless/qca,ath9k.txt) for qcom,ath9k bindings. However, someone mentioned that it should be the reg address instead of device/vendor IDs...so which way should it really be? Is the syntax in the docs not consistent or not preferred?

https://patchwork.kernel.org/patch/9317949/
search in page "unit address"

Reply:
@mpratt14 Adrian is almost right on this one, the example is wrong

some background as to what I've done wrong in the example:
- the unit-address is what goes after the "@" sign
- this is important to ensure that the nodes are unique after compiling to .dtb
- it should contain all the address-cells (for example: ``ethernet-phy@4`` also has ``reg = <4>``)
- when there are multiple address-cells (``#address-cells = <3>; and #size-cells = <2>;`` for `&pcie` in `ar724x.dtsi`) then we need to use all all addresses in after the "@" sign
  - for example: ``reg = <0xa 0xb 0xc 0xd 0xe>`` will have ``device@a,b,c`` (0xd and 0xe are the size-cells)
- for an example you can check upstream ``arch/mips/boot/dts/img/boston.dts`` - although I'm surprised to see that ``arch/arm/boot/dts/bcm47189-tenda-ac9.dts`` only follows this pattern for ``bridge@0,0,0`` but not for ``wifi@0,1,0``
- when in doubt, clarify with the upstream device-tree maintainers

Reply:
OK I also see that address-cells is 3 in the dtsi, so based on that I'll do `wifi@0,0,0` for both PRs then....will push later...